### DISCUSSION TITLE
PSA: Arch isn't thread safe

### DISCUSSION THREAD

**Discussion by dtandersen** [genaray/Arch#142] [Q&A] [answered]
If you're using Xunit for tests you can add this:

[assembly: CollectionBehavior(DisableTestParallelization = true)]

It will save you many hours of hair pulling!

50% of the time worlds were assigned the same World Id and Arch went nuts.

I wonder if https://github.com/genaray/Arch/issues/123 could be related?

**Comment by genaray** [comment, marked as answer]
Thats right, thanks for pointing out! 
Arch itself is not thread-safe ^^
